modules/sdk/src/main/scala/org/tessellation/sdk/domain/fork/ForkInfoQueue.scala

```diff
@@ -0,0 +1,23 @@
+package org.tessellation.sdk.domain.fork
+
+import cats.syntax.eq._
+
+import scala.collection.immutable.Queue
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosInt
+
+case class ForkInfoQueue(value: Queue[ForkInfo], maxSize: PosInt) {
+  def add(entry: ForkInfo): ForkInfoQueue = {
+    val updated = (if (value.size === maxSize) value.tail else value) :+ entry
+
+    ForkInfoQueue(updated, maxSize)
+  }
+}
+
+object ForkInfoQueue {
+  def empty(maxSize: PosInt): ForkInfoQueue = ForkInfoQueue(value = Queue.empty, maxSize = maxSize)
+
+  def fromList(entries: List[ForkInfo], maxSize: PosInt): ForkInfoQueue =
+    entries.foldLeft(ForkInfoQueue.empty(maxSize)) { case (acc, entry) => acc.add(entry) }
+}
```

Comment:
1. I don't think you should name your class based on the internal implementation. Also, your constructor allows the caller to create your object with an invalid state, a queue with more than `maxSize` elements.
2. Why are you passing in a Queue? Isn't that an internal (hidden) implementation detail? Your constructor should probably be equivalent to calling `empty`.
3. Is this really a `case class` ? Feels more like `class` to me.
4. `fromList` is too specific. Look at the interface for `Queue` and you can reduce the constraint on the inpur argument and probably also simplify your implementation.

Reply:
1. What do you recommend?
2. The case class has been made a sealed case class and the only way to make it is with the `empty` and `from` functions.
3. Implementing it as a class gave me too many issues. Making it a sealed case class seems to resolve your other concerns.
4. Generalized to `from` that works with any `IterableOnce`.

Reply:
> Implementing it as a class gave me too many issues.
@cngo-github can you describe some of them? `case class` is kind of sugar syntax that allows you to pattern match and I doubt you would use pattern matching for the queue

> I don't think you should name your class based on the internal implementation.
@sdavidp IMO it depends what level of abstraction you need. if that's a more specialized queue, then it's fine to name it a Queue. if it's a domain part, then it leaks the implementation in the name indeed

Reply:
@marcinwadon , I had problems with `eqv` and `show`. This has been resolved in [this comment](https://github.com/Constellation-Labs/tessellation/pull/749#discussion_r1369440934).